services/frontend-service/pkg/cmd/server.go

```diff
@@ -377,21 +381,24 @@ func (p *Auth) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	logger.Wrap(r.Context(), func(ctx context.Context) error {
 		span, ctx := tracer.StartSpanFromContext(ctx, "ServeHTTP")
 		defer span.Finish()
-		var u *auth.User = nil
+		var user *auth.User = nil
 		var source = ""
 		if c.AzureEnableAuth {
-			u = getRequestAuthorFromAzure(r)
+			user = getRequestAuthorFromAzure(r)
 			source = "azure"
 		} else {
-			u = getRequestAuthorFromGoogleIAP(ctx, r)
+			user = getRequestAuthorFromGoogleIAP(ctx, r)
 			source = "iap"
 		}
-		if u != nil {
-			span.SetTag("current-user-name", u.Name)
-			span.SetTag("current-user-email", u.Email)
+		if user == nil {
+			user, _ = auth.ReadUserFromHttpHeader(ctx, r)
+		}
+		if user != nil {
+			span.SetTag("current-user-name", user.Name)
```

Comment:
This doens't look related to your story. Maybe put this in an extra PR?

Reply:
It is subtle, but it is related:

Before we only read the user from google/azure. Now we read it from the http header too - otherwise it would be always empty for the /release endpoint.

It is now an issue, because I want to pass along the author from an http endpoint (release) in the frontend-service.
Since we asked users to provide the header (here https://github.com/freiheit-com/kuberpult/blob/main/README.md#releasing-a-new-version), I would like to keep that behavior as it was. To achieve that we need to read the http header here.